src/support/timefuncs.c

```diff
@@ -29,30 +29,29 @@ extern "C" {
 #endif
 
 #if defined(_OS_WINDOWS_)
-static double floattime(void)
+DLLEXPORT int gettimeofday(struct timeval *tv, void *tzp)
 {
-    struct timeb tstruct;
-
-    ftime(&tstruct);
-    return (double)tstruct.time + (double)tstruct.millitm/1.0e3;
+    struct _timeb tb;
+    errno_t code = _ftime_s(&tb);
+    if(code == 0)
+    {
+        tv->tv_sec = tb.time;
+        tv->tv_usec = tb.millitm * 1e3;
```

Comment:
`* 1000`, not `1e3` (which converts to floating point, does the multiply, then converts back.

You can just do these assignments unconditionally.  No need for the `code==0` check, since in the worst case it will just write garbage values into `tv`